openwisp_controller/config/base/device.py

```diff
@@ -156,6 +161,15 @@ def save(self, *args, **kwargs):
                 self.key = self.generate_key(shared_secret)
         super().save(*args, **kwargs)
 
+        if self.management_ip != self._initial_management_ip:
+            self.send_is_management_ip_changed_signal()
+        self._initial_management_ip = self.management_ip
+
+    def send_is_management_ip_changed_signal(self):
+        management_ip_changed.send(
+            sender=self.__class__, management_ip=self.management_ip
```

Comment:
The device instance should be sent as well.

Look at the `is_working_changed` signal: it's good to provide the new value and the old value, so in this case I would send `self._initial_management_ip` as `old_management_ip`